libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -697,6 +740,151 @@ namespace {
         return retvalue.size() ? retvalue : "application/octet-stream";
     }
 
+    class PermsCalc {
+        std::map<std::string,std::string> users,groups;
+        std::map<std::string,std::list<std::string>> members;
+        std::map<std::string,int> user_entry;
+        std::multimap<int,std::string> user_perms, group_perms;
+
+        static std::map<int,std::string> perm_names;
+
+        bool is_group( const std::string &gid ) {
+            try {
+                groups.at(gid);
+            }
+            catch (const std::out_of_range&) {
+                return false;
+            }
+            return true;
```

Comment:
instead of relying on exceptions, prefer some variant of the search algorithms (e.g. `std::find`)

Reply:
@d-w-moore research